Sources/MessagingInApp/Gist/Managers/ModalViewManager.swift

```diff
@@ -40,13 +43,15 @@ class ModalViewManager {
             finalPosition = viewController.view.center.y - viewController.view.bounds.height
         }
 
-        UIView.animate(withDuration: 0.3, delay: 0, options: [.curveEaseIn], animations: {
+        animationRunner.animate(withDuration: 0.3, delay: 0, options: [.curveEaseIn], animations: {
             self.viewController.view.center.y = finalPosition
         }, completion: { _ in
-            UIView.animate(withDuration: 0.1, delay: 0, options: [.curveEaseIn], animations: {
+            self.animationRunner.animate(withDuration: 0.1, delay: 0, options: [.curveEaseIn], animations: {
                 self.viewController.view.backgroundColor = UIColor.black.withAlphaComponent(0.2)
-            }, completion: nil)
-            completionHandler()
+            }, completion: { _ in
+                self.isShowingMessage = true
+                completionHandler()
```

Comment:
`self.isShowingMessage = true` is only set until after all animations are done running.